### PR TITLE
gh actions

### PR DESCRIPTION
**Pull request by bwagner5**
*Issue #, if available:*
N/A

*Description of changes:*
 - Move to github actions

https://github.com/bwagner5/aemm-test/runs/1444326486?check_suite_focus=true

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,185 @@
+name: EC2 Metadata Mock CI and Release
+
+on: [push, pull_request, workflow_dispatch]
```
Comment (brycahta) [inline .github/workflows/ci.yaml:3]: Still doing some reading, but is `workflow_dispatch` here so we can kick off these jobs from the repo without any changes (i.e. commits, prs, tags, etc.)?

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+  GITHUB_TOKEN: ${{ secrets.EC2_BOT_GITHUB_TOKEN }}
+  DOCKERHUB_USERNAME: ${{ secrets.DOCKERHUB_USERNAME }}
+  DOCKERHUB_PASSWORD: ${{ secrets.DOCKERHUB_PASSWORD }}
+  DOCKERHUB_TOKEN: ${{ secrets.DOCKERHUB_TOKEN }}
+
+jobs:
+
+  buildAndTest:
+    name: Build and Test
+    runs-on: ubuntu-20.04
```
Comment (brycahta) [inline .github/workflows/ci.yaml:23]: any particular reason why we're locking in this ubuntu version vs. running on `ubuntu-latest`?

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+      with:
+        go-version: ${{ env.DEFAULT_GO_VERSION }}
+
+    - name: Check out code into the Go module directory
+      uses: actions/checkout@v2
+
+    - name: Build Docker Images Windows
+      run: choco install make && RefreshEnv.cmd && make build-docker-images-windows
+    
+  release:
```
Comment (brycahta) [inline .github/workflows/ci.yaml:82]: Can do the [If at the job level](https://github.blog/changelog/2019-10-01-github-actions-new-workflow-syntax-features/) so we don't need it for each step

File: .github/workflows/ci.yaml [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+      run: make release-github
+
+    - name: Release Docker Linux
+      if: ${{ env.IS_RELEASE == true }}
+      run: make release-docker-linux
+    
+  releaseWindows:
+    name: Release Windows
+    runs-on: windows-2019
+    needs: [buildAndTest]
```
Comment (brycahta) [inline .github/workflows/ci.yaml:112]: should this also needs `buildWindows` before kicking off? If that job fails, then this one will for sure

File: .github/workflows/ci.yaml [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+      uses: actions/setup-go@v2
+      with:
+        go-version: ${{ env.DEFAULT_GO_VERSION }}
+
+    - name: Check out code into the Go module directory
+      uses: actions/checkout@v2
+    
+    - name: Helm Chart Tests
+      if: ${{ env.SHOULD_TEST_HELM == true }}
+      run: test/helm/chart-test.sh -i -k ${{ matrix.k8sVersion }}
```
Comment (brycahta) [inline .github/workflows/ci.yaml:185]: k8sVersions lookin clean 👍

**Review by brycahta — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (bwagner5) [reply .github/workflows/ci.yaml:3]: yep, for manual kick off

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (bwagner5) [reply .github/workflows/ci.yaml:23]: I wanted this to be deterministic, I'm not opposed to using latest, but that was the reasoning.

File: .github/workflows/ci.yaml (hunk 3 above)
Comment (bwagner5) [reply .github/workflows/ci.yaml:82]: ohhhhhhh I didn't see that! I'll change

File: .github/workflows/ci.yaml (hunk 4 above)
Comment (bwagner5) [reply .github/workflows/ci.yaml:112]: ah yeah

**Review by brycahta — APPROVED** [review]
Looks good!! 🚀

**Pull request merged**